### ISSUE TITLE
[icetray] Automatic OutBox breaks TrashCan (Trac #1998)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1990]
The automatic OutBox added in r2868 also adds an OutBox to the TrashCan module, resulting in:

```text
FATAL (I3Module): Module "TrashCan_0000" outbox "OutBox" not connected (I3Module.cxx:498 in void I3Module::EnforceOutBoxesConnected() const)
```
Either back it out or make `I3Module::EnforceOutBoxesConnected()` more clever.

Migrated from https://code.icecube.wisc.edu/ticket/1998
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:53", 
    "description": "The automatic OutBox added in r2868 also adds an OutBox to the TrashCan module, resulting in:\n{{{\nFATAL (I3Module): Module \"TrashCan_0000\" outbox \"OutBox\" not connected (I3Module.cxx:498 in void I3Module::EnforceOutBoxesConnected() const)\n}}}\nEither back it out or make `I3Module::EnforceOutBoxesConnected()` more clever.", 
    "reporter": "jvansanten", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067233566919", 
    "component": "combo core", 
    "summary": "[icetray] Automatic OutBox breaks TrashCan", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2017-05-05T20:28:27", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170507T22:26:47: david.schultz commented:

> I tried the simple solution of just overriding `Configure()` for the `TrashCan` module.  This works in most cases.
> 
> Where it fails:
> - create one dummy module without a `Configure()`, and without an `OutBox` set up
> - call `tray.Execute()`
> - auto-trashcan is skipped, because there is no outbox yet
> - default `Configure()` is called, adding an outbox
> - `EnforceOutBoxesConnected()` is called, which fails
> 
> This worked before because the simple tests didn't need any connecting outboxes or trashcans.
> 
> So, two easy options:
> 
> a. Delay the auto-trashcan detection until after `Configure()` is called
> b. Try hoisting the outbox auto-creation earlier in the steps

**Comment by icecube-trac** [comment]
Trac update at 20170507T23:19:40: david.schultz commented:

> Option 1 seems to work.  All tests pass.
> 
> It has one side effect: you can now add new modules during the `Configure()` step.  I don't think this is actually a problem though.

**Comment by icecube-trac** [comment]
Trac update at 20170507T23:20:47: 
* david.schultz changed resolution from "" to "fixed"
* david.schultz commented:

> In changeset 2872:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2872"
> override Configure in the TrashCan module, and move the auto-adding of the trashcan after the Configure step. closes #1990
> ```


* david.schultz changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T20:55:42: nwhitehorn commented:

> In changeset 168805:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168805"
> override Configure in the TrashCan module, and move the auto-adding of the trashcan after the Configure step. closes #1990
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:53: 
* olivas commented:

> Milestone May 2017 deleted


* olivas changed milestone from "May 2017" to ""
